composer.json

```diff
@@ -19,6 +19,7 @@
         "php": ">=5.5",
         "php-http/client-implementation": "^1.0",
         "php-http/message-factory": "^1.0.2",
+        "php-http/client-common": "^1.1",
         "php-http/plugins": "^1.0",
```

Comment:
if we require both the new and the old plugins now, we don't need the flexible code you introduce above.

but i think to avoid BC breaks, we can indeed depend on both.